# Data Beliefs Vs Preferences for Remote, Camera, and Recordings

Using spearman's correlation coefficient (because we are measuring the correlation of mostly ordinal data) and the metric of 

    < 20    "very weak - no correlation"
    .20-.39 “weak”
    .40-.59 “moderate”
    .60-.79 “strong”
    .80-1.0 “very strong”

We can tell that there is 

* a moderate correlation between how much participants believe video conferencing collects and how much participants believe the other two categories of software collects (0.47 and 0.45 respectively)
* a strong correlation between how much participants believe LMS collects and how much participants believe Proctoring software collects (0.61)
* **a weak negative correlations between belief in how much video conferencing software collects and whether students wanted lectures recorded (-0.26). Also a weak correlation as to whether they believed that the data was sent to instructors (0.24) No correlation to whether they want the camera on nor whether they wanted remote (-0.01 and -0.09 respectively)**
* moderate correlation between whether they believed that video conferencing data was sent to instructors vs whether LMS data was as well as whether proctoring data was. (0.48 and 0.47 respectively) wrt LMS data vs proctoring data is also moderate (0.41)
* not much correlation between whether they wanted the camera on nor if they wanted remote classes with anything else

For the pairs that do not have strong correlation according to spearman this does not mean that there is indeed no/weak correlation but rather that there is no/weak monotonic correlation

In [312]:
data_copy.corr(method='spearman', min_periods=1)

beliefsVidConId  beliefsDataSentVidCon  beliefsLMSId  \
beliefsVidConId                1.000000               0.240772      0.473345   
beliefsDataSentVidCon          0.240772               1.000000      0.026091   
beliefsLMSId                   0.473345               0.026091      1.000000   
beliefsDataSentLMS             0.137432               0.481306      0.249059   
beliefsProctorId               0.456845               0.142714      0.617363   
beliefsDataSentProctor         0.032123               0.473015      0.049733   
ifWantCameraOn                -0.008632              -0.044089     -0.070213   
ifWantRemote                  -0.099356               0.181481      0.059147   
ifWantRecording               -0.263384              -0.202774     -0.205922   

                        beliefsDataSentLMS  beliefsProctorId  \
beliefsVidConId                   0.137432          0.456845   
beliefsDataSentVidCon             0.481306          0.142714   
beliefsLMSId                      0.249059          0.617363   
beliefsDataSentLMS                1.000000          0.055549   
beliefsProctorId                  0.055549          1.000000   
beliefsDataSentProctor            0.414213          0.195004   
ifWantCameraOn                    0.062841         -0.167775   
ifWantRemote                     -0.030877          0.104617   
ifWantRecording                  -0.203707         -0.061922   

                        beliefsDataSentProctor  ifWantCameraOn  ifWantRemote  \
beliefsVidConId                       0.032123       -0.008632     -0.099356   
beliefsDataSentVidCon                 0.473015       -0.044089      0.181481   
beliefsLMSId                          0.049733       -0.070213      0.059147   
beliefsDataSentLMS                    0.414213        0.062841     -0.030877   
beliefsProctorId                      0.195004       -0.167775      0.104617   
beliefsDataSentProctor                1.000000        0.003001      0.135125   
ifWantCameraOn                        0.003001        1.000000      0.048074   
ifWantRemote                          0.135125        0.048074      1.000000   
ifWantRecording                      -0.041888       -0.040734      0.104692   

                        ifWantRecording  
beliefsVidConId               -0.263384  
beliefsDataSentVidCon         -0.202774  
beliefsLMSId                  -0.205922  
beliefsDataSentLMS            -0.203707  
beliefsProctorId              -0.061922  
beliefsDataSentProctor        -0.041888  
ifWantCameraOn                -0.040734  
ifWantRemote                   0.104692  
ifWantRecording                1.000000

# \# of non-unsure items particpant believes is collected vs preferences

Response to:
> There’s also no discussion regarding these quantitative results in the Results and Discussion section. Looking at the survey questionnaire, the questions regarding students’ beliefs about the different types of data collected by the different software can potentially be used to predict students’ preference for attending classes remotely vs. in-person.

Though we didnt find too much correlation wrt our Spearman test, we can do some more tests on this. I think One way ANOVA is appropriate here as we have a discrete variable (prefernces for remote/recording/camera) and continuous (# of data collected)

In [398]:
a = [c for c in data_copy.columns if 'ifWant' in c]
b = [c for c in data_copy.columns if 'beliefs' in c and 'Id' in c]

table = {'beliefsVidConId': [], 'beliefsLMSId': [], 'beliefsProctorId': []}
#table['beliefs_vidcon'].append(4)
#table
for cat in b:
    for item in a:
        F, p = ss.f_oneway(data_copy[data_copy[item] == 0][cat],
                  data_copy[data_copy[item] == 1][cat],
                  data_copy[data_copy[item] == 2][cat],
                  data_copy[data_copy[item] == 3][cat],
                  data_copy[data_copy[item] == 4][cat])
        table[cat].append((p,F))
pd.DataFrame(data=table, index=a)

beliefsVidConId  \
ifWantCameraOn     (0.039949095666883164, 2.58847110125591)   
ifWantRemote       (0.5592503795873202, 0.7508270200672893)   
ifWantRecording  (0.004840248869209463, 3.9311082360639475)   

                                               beliefsLMSId  \
ifWantCameraOn     (0.5899299387418798, 0.7050303334538528)   
ifWantRemote      (0.7449654838947286, 0.48739931817161364)   
ifWantRecording  (0.00037614441177719814, 5.55462752414772)   

                                         beliefsProctorId  
ifWantCameraOn   (0.14338666238173736, 1.748820406650797)  
ifWantRemote     (0.4843732730874074, 0.8693864491216069)  
ifWantRecording  (0.2894176262381812, 1.2596525004596557)

We find:
* Statistical significance wrt # of categories participant believed both video conference software and LMS collected vs whether recording is wanted
* \# of categories participant believed video conferencing software collected and whether they beleived the camera should be on

In [360]:
#data_copy[data_copy["ifWantCameraOn"] == 0]["beliefsVidConId"]
F, p = ss.f_oneway(data_copy[data_copy["ifWantCameraOn"] == 0]["beliefsVidConId"],
                  data_copy[data_copy["ifWantCameraOn"] == 1]["beliefsVidConId"],
                  data_copy[data_copy["ifWantCameraOn"] == 2]["beliefsVidConId"],
                  data_copy[data_copy["ifWantCameraOn"] == 3]["beliefsVidConId"],
                  data_copy[data_copy["ifWantCameraOn"] == 4]["beliefsVidConId"])
F, p

(2.58847110125591, 0.039949095666883164)

In [367]:
F, p = ss.f_oneway(data_copy[(data_copy["ifWantCameraOn"] == 0) | (data_copy["ifWantCameraOn"] == 1)]["beliefsVidConId"],
                  data_copy[data_copy["ifWantCameraOn"] == 2]["beliefsVidConId"],
                  data_copy[(data_copy["ifWantCameraOn"] == 3) | (data_copy["ifWantCameraOn"] == 4)]["beliefsVidConId"])
F, p

(0.9839846335561695, 0.3766281118395992)

So here it is interesting that when we run ANOVA on camera preferences vs # of categories participant believes is being collected by video conferencing software as is, we get p = 0.04, F = 2.59, however when we roll up the responses to correspond to just negative, neutral, and positive preference, we get a lot less statistical significance.

# Beliefs whether data is sent to instructor vs preferences

In [473]:
a = [c for c in data_copy.columns if 'ifWant' in c]
b = categorical_cols

table = []

for i in a:
    row = []
    for j in b:
        pair = data_copy[[i,j]].applymap(lambda x: np.nan if x < 0 else x)
        pair = pair[~pair.isnull().any(axis=1)]
        chi, p, df, ex = ss.chi2_contingency(pd.crosstab(pair[i], pair[j]))
        row.append((p,df))
    table.append(row)

table = pd.DataFrame(data = table, columns=b, index = a)
table

beliefsDataSentVidCon         beliefsDataSentLMS  \
ifWantCameraOn   (0.6301106051320201, 4)     (0.928875210318349, 4)   
ifWantRemote     (0.4145279548642756, 4)  (0.017134640093821697, 4)   
ifWantRecording  (0.9782777131606106, 4)    (0.8168866828650898, 4)   

                    beliefsDataSentProctor  
ifWantCameraOn   (0.019745184565358766, 4)  
ifWantRemote      (0.21532138340191687, 4)  
ifWantRecording    (0.4717049550176151, 4)

# Presense of Not Sure vs Preferences

Here we can do Mann Whitney or Chi Squared. Since we use chi squared prior and will for the next few tests, for consistency we will use Chi Squared

In [474]:
data_copy[categorical_cols] = data[categorical_cols]#.applymap(lambda x: np.nan if x < 0 else x)
data_copy[categorical_cols] = data_copy[categorical_cols].applymap(lambda x: 1 if x > 0 else x)
#data_copy[categorical_cols]

In [475]:
a = [c for c in data_copy.columns if 'ifWant' in c]
b = categorical_cols

table = []

for i in a:
    row = []
    for j in b:
        pair = data_copy[[i,j]].applymap(lambda x: np.nan if x < 0 else x)
        pair = pair[~pair.isnull().any(axis=1)]
        #print(pd.crosstab(pair[i], pair[j]))
        chi, p, df, ex = ss.chi2_contingency(pd.crosstab(pair[i], pair[j]))
        row.append((p,df))
    table.append(row)

table = pd.DataFrame(data = table, columns=b, index = a)
table

beliefsDataSentVidCon         beliefsDataSentLMS  \
ifWantCameraOn   (0.6301106051320201, 4)     (0.928875210318349, 4)   
ifWantRemote     (0.4145279548642756, 4)  (0.017134640093821697, 4)   
ifWantRecording  (0.9782777131606106, 4)    (0.8168866828650898, 4)   

                    beliefsDataSentProctor  
ifWantCameraOn   (0.019745184565358766, 4)  
ifWantRemote      (0.21532138340191687, 4)  
ifWantRecording    (0.4717049550176151, 4)

**So there is some sort of relationship between whether participants want remote classes and whether they believe data is sent to instructors both a relationship between wanting remote and a specific belief and whether they were unsure.**
Similarly there seems to be some sort of relationship between whether participants want mandatory camera and their beliefs of whether proctoring data is sent to instructors. I'm a little dubious about this one since most of the time we while we associate proctor with camera, the relationship doesnt go the other way around as camera is more associated with video conferencing software.

# Presence of specific data category in beliefs vs preferences

In [545]:
a = [c for c in data_copy.columns if 'ifWant' in c]
b = [c for c in data_copy.columns if 'Id' in c]
vcTable = []
lmsTable = []
proctorTable = []
for x in range(1, 14):
    if x != 4:
        temp = data.copy(deep = True)
        temp[b] = temp[b].applymap(lambda l: -1 if (pd.notnull(l) is False) else (1 if (str(x) in l or " "+str(x) in l) else 0))
        temp[a] = temp[a].applymap(lambda l: np.nan if l < 0 else l)
        temp[b] = temp[b].applymap(lambda l: np.nan if l < 0 else l)
        
        vcrow = []
        lmsrow = []
        proctorrow = []
        for i in a:
            vcpair = temp[[i,"beliefsVidConId"]]
            LMSpair = temp[[i,"beliefsLMSId"]]
            proctorpair = temp[[i,"beliefsProctorId"]]
            
            vcpair = vcpair[~vcpair.isnull().any(axis=1)]
            LMSpair = LMSpair[~LMSpair.isnull().any(axis=1)]
            proctorpair = proctorpair[~proctorpair.isnull().any(axis=1)]
            
            chi, p, df, ex = ss.chi2_contingency(pd.crosstab(vcpair[i], vcpair["beliefsVidConId"]))
            vcrow.append((p,df))
            chi, p, df, ex = ss.chi2_contingency(pd.crosstab(LMSpair[i], LMSpair["beliefsLMSId"]))
            lmsrow.append((p,df))
            chi, p, df, ex = ss.chi2_contingency(pd.crosstab(proctorpair[i], proctorpair["beliefsProctorId"]))
            proctorrow.append((p,df))
            
        vcTable.append(vcrow)
        lmsTable.append(lmsrow)
        proctorTable.append(proctorrow)
vcTable = pd.DataFrame(data = vcTable, columns=a, index = [c for c in range(1,14) if c != 4])
lmsTable = pd.DataFrame(data = lmsTable, columns=a, index = [c for c in range(1,14) if c != 4])
proctorTable = pd.DataFrame(data = proctorTable, columns=a, index = [c for c in range(1,14) if c != 4])
        
        #print(temp[b].value_counts())
        # 1 if category present 0 if not

Whether belief that any specific category is collected correlates with any preferences

From the below tables we can tell that:
* Beliefs Hardware Specs collected for video conferencing software and if wanting camera on seem to have a relationship (I dont see a connection here though
* **Belief that video conferencing collects personal file and data have a relation with whether want camera, recording and remote** (p = 0.05, 0.027, 0.004 respectively)
* **Belief that video conferencing collects sensitive AND personal information have a relation with whether want camera, and recording** (p = 0.0002, 0.01, 0.017, 0.0003 respectively)
* LMS collects keystrokes vs camera on -- once again this might be a coincidence (p = 0.013)
* LMS collects sensitive info vs recording (p = 0.017)

In [546]:
vcTable

ifWantCameraOn               ifWantRemote  \
1       (0.2400396810941412, 4)     (0.752829498984052, 4)   
2       (0.5939558870261725, 4)   (0.04536565572271165, 4)   
3       (0.0360922163486274, 4)    (0.4385363251546247, 4)   
5      (0.07910093233627037, 4)    (0.5043063296746515, 4)   
6      (0.43114046396457506, 4)    (0.6497100835989296, 4)   
7       (0.8412930961314267, 4)   (0.24545370064691505, 4)   
8      (0.05315214879413005, 4)  (0.027997005159418477, 4)   
9      (0.09880289446167267, 4)    (0.3385385907931029, 4)   
10     (0.42808262656827545, 4)    (0.1869006954961551, 4)   
11  (0.00018357973573476655, 4)    (0.0980143891533722, 4)   
12      (0.9771121795141983, 4)     (0.707373189872599, 4)   
13    (0.017225984206168946, 4)    (0.7587027523197031, 4)   

                ifWantRecording  
1       (0.2633998508264018, 4)  
2       (0.4818134321314139, 4)  
3      (0.36241110110040786, 4)  
5        (0.635605408274249, 4)  
6      (0.17664927503990427, 4)  
7       (0.5188689687715833, 4)  
8     (0.004717505680557055, 4)  
9       (0.2629938942722825, 4)  
10     (0.07347358642218184, 4)  
11    (0.013035994824437116, 4)  
12      (0.7638089720979258, 4)  
13  (0.00037674607336918395, 4)

In [547]:
lmsTable

ifWantCameraOn              ifWantRemote  \
1     (0.5471068913792954, 4)   (0.9744332520723652, 4)   
2     (0.9563462124489982, 4)  (0.39036994112170453, 4)   
3    (0.14716459982643623, 4)  (0.40820648258013537, 4)   
5     (0.3545827420143653, 4)   (0.5030004848878702, 4)   
6   (0.022372482927146383, 4)  (0.30821096417280414, 4)   
7     (0.8025618454564092, 4)  (0.33910284155234677, 4)   
8     (0.7441689290776747, 4)  (0.20792458657059754, 4)   
9   (0.013018270499219917, 4)   (0.1446923571817413, 4)   
10    (0.6811073767522355, 4)  (0.37839792433359687, 4)   
11   (0.12321763505580469, 4)  (0.29853731105845754, 4)   
12    (0.7088874285866879, 4)   (0.3920828790542405, 4)   
13    (0.9206321365341275, 4)   (0.5921882336534358, 4)   

                ifWantRecording  
1      (0.06326768732123013, 4)  
2       (0.9291020992391266, 4)  
3       (0.6364757125245643, 4)  
5       (0.8285388953530799, 4)  
6        (0.531197551348012, 4)  
7      (0.23169435209062575, 4)  
8      (0.16155562415694083, 4)  
9      (0.13773471999422132, 4)  
10  (0.00032535987793234824, 4)  
11    (0.017166310026881078, 4)  
12     (0.11510374931479682, 4)  
13      (0.1171947392963581, 4)

In [548]:
proctorTable

ifWantCameraOn               ifWantRemote  \
1      (0.4507589235397026, 4)    (0.3744661300715102, 4)   
2     (0.38731718915440233, 4)    (0.5454662135114297, 4)   
3     (0.14229955324830992, 4)    (0.9209288370544902, 4)   
5     (0.24468272933138047, 4)   (0.34371082219814064, 4)   
6     (0.23476783816866742, 4)    (0.5244616911774664, 4)   
7     (0.06656461801456201, 4)    (0.8539959884381626, 4)   
8     (0.12622845655710233, 4)    (0.4131804090178436, 4)   
9     (0.04089381022077005, 4)    (0.7759417386114921, 4)   
10    (0.01704245067110891, 4)    (0.8593963130063793, 4)   
11  (9.858702508563383e-05, 4)  (0.050313810605321245, 4)   
12    (0.03943445473379904, 4)    (0.6943823123921347, 4)   
13   (0.004221379129471842, 4)    (0.3681614051867047, 4)   

             ifWantRecording  
1    (0.7811663733302577, 4)  
2    (0.9155257075989075, 4)  
3    (0.9139618958323698, 4)  
5     (0.859012684554305, 4)  
6   (0.40690054816868304, 4)  
7    (0.6459142607704549, 4)  
8   (0.12725967801988725, 4)  
9     (0.892442976950213, 4)  
10  (0.41399042979878586, 4)  
11   (0.1196826746966147, 4)  
12   (0.8816708238007884, 4)  
13   (0.0920928195385777, 4)

In [341]:
#sb.jointplot(x="ifWantCameraOn", y="beliefsVidConId", data=data_copy, alpha=0.2)
#sb.jointplot(x="ifWantCameraOn", y="beliefsLMSId", data=data_copy, alpha=0.2)
#sb.jointplot(x="ifWantCameraOn", y="beliefsProctorId", data=data_copy, alpha=0.2)

# RUN THE BELOW FIRST OR NOTHING WILL WORK

In [231]:
file = "data.csv"
dbfile = "data.db"
qualfile = "qualStats.csv"

In [232]:
import numpy as np
import pandas as pd
import scipy.stats as ss

import matplotlib.pyplot as plt
import seaborn as sb

In [292]:
data = pd.read_csv(file)

In [293]:
cols = [c for c in data.columns if 'beliefs' in c]
cols.extend(["ifWantCameraOn", "ifWantRemote", "ifWantRecording"])
data = data.filter(cols, axis = 1)
data.drop([c for c in cols if 'DataSent'in c and 'Id' in c],inplace = True, axis = 1)
print(data.columns)

Index(['beliefsVidConId', 'beliefsVidCon', 'beliefsDataSentVidCon',
       'beliefsLMSId', 'beliefsLMS', 'beliefsDataSentLMS', 'beliefsProctorId',
       'beliefsProctor', 'beliefsDataSentProctor', 'ifWantCameraOn',
       'ifWantRemote', 'ifWantRecording'],
      dtype='object')


In [294]:
colBeliefs = [c for c in data.columns if 'Id' in c]
#data[colBeliefs] = data[colBeliefs].apply(split)
data[colBeliefs] = data[colBeliefs].astype('string')
#print(data[colBeliefs].astype('string').dtypes)
#print(data[colBeliefs])
data[colBeliefs] = data[colBeliefs].apply(lambda x: x.str.split(','))
#print(data[colBeliefs])


In [296]:
ordinal_cols = [c for c in data.columns if 'ifWant' in c]
data[ordinal_cols] = data[ordinal_cols].astype(pd.api.types.CategoricalDtype(ordered = True, categories=['Definitely No', 'No', 'Might or Might Not', 'Yes', 'Definitely Yes']))
data[ordinal_cols] = data[ordinal_cols].apply(lambda x : x.cat.codes)

In [297]:
categorical_cols = [c for c in data.columns if 'beliefsData' in c]
data[categorical_cols] = data[categorical_cols].astype(pd.api.types.CategoricalDtype(ordered = True, categories=["I'm not sure", 'No', 'Yes. Some of it', 'Yes. All of it']))
data[categorical_cols] = data[categorical_cols].apply(lambda x : x.cat.codes)

(-1) I'm not sure

(1)  Video/ Audio Recordings

(2)  IP Address

(3)  Hardware and System Specifications (eg operating system, type of device, internet service provider, installed peripherals such as monitors or keyboards, etc)

(5)  Browser Specification

(6)  Open Applications or Browser Tabs

(7)  Installed Browser Plugins

(8)  Personal Files/ Documents

(9)  Keystrokes/ Mouse Activity

(10) Facial Movements

(11) Sensitive Information (information that may result in harm or discrimination eg race, health information, political beliefs, etc)

(12) Behavioral Information (information about how you use the software eg. how much time you spend on the software, what features you use

(13) Personal Information (information that can be used to identify you eg. phone number, address, employer, SSN. For the purposes of this survey your name and email address are not considered personal information)

\* accidentally skipped (4) while transforming answers to corresponding number so there is no (4). There were only 13 choices for these questions

In [461]:
data_copy = data.copy(deep = True)

Count the # of items not including "I'm not sure"

In [462]:
data_copy[colBeliefs] = data_copy[colBeliefs].applymap(lambda x: len(x)-1 if '-1' in x else len(x), na_action='ignore')

In [463]:
data_copy[colBeliefs] = data_copy[colBeliefs].fillna(0)

In [464]:
#data_copy[categorical_cols]

In [465]:
data_copy[categorical_cols] = data_copy[categorical_cols].applymap(lambda x: np.nan if x < 1 else x)

In [466]:
#data_copy